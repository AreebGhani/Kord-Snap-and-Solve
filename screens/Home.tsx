import React from "react";
import { StatusBar } from "expo-status-bar";
import { View } from "react-native";
import Card from "@/components/Card";

type Props = {};

const Home: React.FC<Props> = (props: Props) => {
  return (
    <View className="bg-gray-900 flex flex-col flex-1 justify-center items-center">
      <StatusBar style="light" />
      <Card />
    </View>
  );
};

export default Home;
