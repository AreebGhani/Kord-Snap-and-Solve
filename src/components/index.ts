/* This code snippet is exporting components from different files in a TypeScript project. Each line is
exporting a default component from a specific file using the `export { default as ComponentName }
from 'filePath';` syntax. */
export { default as Card } from '@/src/components/Card';
