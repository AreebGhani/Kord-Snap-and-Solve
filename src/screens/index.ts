/* This code snippet is exporting screens from different files in a TypeScript project. Each line is
exporting a default screen from a specific file using the `export { default as ScreenName }
from 'filePath';` syntax. */
export { default as HomeScreen } from '@/src/screens/HomeScreen';
