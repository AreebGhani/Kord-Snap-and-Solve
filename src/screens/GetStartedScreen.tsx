/**
 * The GetStartedScreen component renders a view with a status bar, a card component, and a button
 * linked to the login page.
 */
import { Button, View } from 'react-native';
import { Link } from 'expo-router';

import { Card } from '@/src/components';

const GetStartedScreen = () => (
  <View className="bg-gray-900 flex flex-col flex-1 justify-center items-center">
    <Card text="Get Started Screen" />
    <View className="my-10" />
    <Link href="/login" push asChild>
      <Button title="Login your account" />
    </Link>
  </View>
);

export default GetStartedScreen;
