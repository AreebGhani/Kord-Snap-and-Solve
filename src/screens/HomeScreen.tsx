/**
 * The Home screen renders a View with a background color, StatusBar, and a Card component in a
 * React Native application.
 */
import { StatusBar } from 'expo-status-bar';
import { View } from 'react-native';

import { Card } from '@/src/components';

const HomeScreen = () => (
  <View className="bg-gray-900 flex flex-col flex-1 justify-center items-center">
    <StatusBar style="light" />
    <Card />
  </View>
);

export default HomeScreen;
