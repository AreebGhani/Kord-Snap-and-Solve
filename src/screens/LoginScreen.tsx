/**
 * The `LoginScreen` component renders a login screen with a card displaying "Login Screen" text and a
 * button to create an account.
 */
import { Link } from 'expo-router';
import { View, Button } from 'react-native';

import { Card } from '@/src/components';

const LoginScreen = () => (
  <View className="bg-gray-900 flex flex-col flex-1 justify-center items-center">
    <Card text="Login Screen" />
    <View className="my-10" />
    <Link href="/home" replace asChild>
      <Button title="Home" />
    </Link>
    <View className="my-10" />
    <Link href="/signup" replace asChild>
      <Button title="Create your account" />
    </Link>
  </View>
);

export default LoginScreen;
