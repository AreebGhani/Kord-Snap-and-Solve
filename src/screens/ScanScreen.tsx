import { View } from 'react-native';

import { Card } from '@/src/components';

const ScanScreen = () => (
  <View className="bg-gray-900 flex flex-col flex-1 justify-center items-center">
    <Card text="Scan Screen" />
    <View className="my-10" />
  </View>
);

export default ScanScreen;
