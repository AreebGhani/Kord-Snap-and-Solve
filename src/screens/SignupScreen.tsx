/**
 * The SignupScreen component in TypeScript React renders a signup screen with a card component and a
 * button to navigate to the login screen.
 */
import { Link } from 'expo-router';
import { View, Button } from 'react-native';

import { Card } from '@/src/components';

const SignupScreen = () => (
  <View className="bg-gray-900 flex flex-col flex-1 justify-center items-center">
    <Card text="Signup Screen" />
    <View className="my-10" />
    <Link href="/login" replace asChild>
      <Button title="Already have an account" />
    </Link>
  </View>
);

export default SignupScreen;
