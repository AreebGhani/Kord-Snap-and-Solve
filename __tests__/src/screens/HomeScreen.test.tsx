/* This code snippet is a test file written in TypeScript for a React application. Here's a breakdown
of what it does: */
import React from 'react';
import renderer from 'react-test-renderer';

import { HomeScreen } from '@/src/screens';

// Mock StatusBar to prevent rendering issues during testing
jest.mock('expo-status-bar', () => ({
  StatusBar: 'StatusBar',
}));

// Mock Card Component
jest.mock('@/src/components/Card', () => 'MockedCard');

describe('Home Screen', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
