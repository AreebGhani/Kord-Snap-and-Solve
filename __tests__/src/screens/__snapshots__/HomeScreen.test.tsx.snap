// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#111827",
      },
      {
        "display": "flex",
      },
      {
        "flexDirection": "column",
      },
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "justifyContent": "center",
      },
      {
        "alignItems": "center",
      },
    ]
  }
>
  <StatusBar
    style="light"
  />
  <MockedCard />
</View>
`;
